### ISSUE TITLE
Error trying to compile the simjectExample tweak 

### ISSUE THREAD

**Issue by raulsaeed** [akemin-dayo/simject#83]
❯ make setup
==> Cleaning…
==> Notice: Build may be slow as Theos isn’t using all available CPU cores on this computer. Consider upgrading GNU Make: https://theos.dev/docs/parallel-building
> Making all for tweak simjectExampleTweak…
==> Preprocessing simjectExampleTweak.x…
==> Compiling simjectExampleTweak.x (arm64)…
==> Linking tweak simjectExampleTweak (arm64)…
ld: warning: -multiply_defined is obsolete
ld: unknown options: -allow_simulator_linking_to_macosx_dylibs
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [/Users/besttech/simject/simjectExampleTweak/.theos/obj/iphone_simulator/arm64/simjectExampleTweak.dylib] Error 1
make[2]: *** [/Users/besttech/simject/simjectExampleTweak/.theos/obj/iphone_simulator/arm64/simjectExampleTweak.dylib] Error 2
make[1]: *** [internal-library-all_] Error 2
make: *** [simjectExampleTweak.all.tweak.variables] Error 2

**Comment by 1vke** [comment]
@raulsaeed 

After installing [Xcode 13.4.1](https://download.developer.apple.com/Developer_Tools/Xcode_13.4.1/Xcode_13.4.1.xip), extracting it, renaming the application to Xcode-13.4.1 to prevent any issues with the current Xcode, and placing it in applications it has seemed to solve my issue.

I can not replicate what I was experiencing to the fullest but it seems like that worked. You might also need to switch your switch your Xcode command line tools around. You can do that via:
`sudo xcode-select -s /Applications/Xcode-13.4.1.app`

To switch back to the most current Xcode command line tools, omit `-13.4.1` in the application path.

Hope this helps.

**Comment by raulsaeed** [comment]
@1vke 
I downloaded the 13.4.1  version of Xcode but it seems it's not compatible  with Sonoma.
The version of Xcode installed on this Mac is not compatible with macOS Sonoma. Download the latest version for free from the App Store.

**Comment by 1vke** [comment]
> @1vke I downloaded the 13.4.1 version of Xcode but it seems it's not compatible with Sonoma. The version of Xcode installed on this Mac is not compatible with macOS Sonoma. Download the latest version for free from the App Store.

Yes it is unsupported. You are supposed to drag it into your applications, not run it. Then when the app is in the folder you should run `sudo xcode-select -s /Applications/Xcode-13.4.1.app`

**Comment by raulsaeed** [comment]
@1vke 
yes that was the case but, now when I compile I get 
`❯ make do
==> Notice: Build may be slow as Theos isn’t using all available CPU cores on this computer. Consider upgrading GNU Make: https://theos.dev/docs/parallel-building
> Making all for tweak simjectExampleTweak…
==> Preprocessing simjectExampleTweak.x…
==> Compiling simjectExampleTweak.x (arm64)…
==> Linking tweak simjectExampleTweak (arm64)…
==> Generating debug symbols for simjectExampleTweak…
==> Stripping simjectExampleTweak (arm64)…
==> Merging tweak simjectExampleTweak…
==> Signing simjectExampleTweak…
error: The specified item could not be found in the keychain.
make[2]: *** [/Users/besttech/simject/simjectExampleTweak/.theos/obj/iphone_simulator/simjectExampleTweak.dylib] Error 1
rm /Users/besttech/simject/simjectExampleTweak/.theos/obj/iphone_simulator/simjectExampleTweak.dylib.3dd3a16a.unsigned
make[1]: *** [internal-library-all_] Error 2
make: *** [simjectExampleTweak.all.tweak.variables] Error 2`

**Comment by 1vke** [comment]
> Making all for tweak simjectExampleTweak…
> ==> Preprocessing simjectExampleTweak.x…
> ==> Compiling simjectExampleTweak.x (arm64)…
> ==> Linking tweak simjectExampleTweak (arm64)…
> ==> Generating debug symbols for simjectExampleTweak…
> ==> Stripping simjectExampleTweak (arm64)…
> ==> Merging tweak simjectExampleTweak…
> ==> Signing simjectExampleTweak…
> error: The specified item could not be found in the keychain.

Perfect! You solved your original issue. 

Your next issue is that you either do not have an apple developer account, or it can't find your developer certificates. 

If you have an Apple Developer account ($99 subscription), follow the instructions in this issue: #43

Unfortunately, I am not sure if there is a way to get around needing an Apple Developer account.

**Comment by raulsaeed** [comment]
@1vke 

ohh is it mandatory to have an apple developer account ?
because I remember asking in some discord server they said that u just need an Apple ID not an paid developer account 
btw I also have an signing service certificate  profile (maple sign is name of the service) can I use that for signing?

**Comment by 1vke** [comment]
> ohh is it mandatory to have an apple developer account ? because I remember asking in some discord server they said that u just need an Apple ID not an paid developer account btw I also have an signing service certificate profile (maple sign is name of the service) can I use that for signing?

I am not sure if you absolutely **need** to have an developer account, but I do know that is what is recommended for this tool, and it works with it. You would have to do your own research on that.

I am not sure if you can use MapleSign certificates. If I were to take a guess, it would be probably a no.

**Comment by PoomSmart** [comment]
What's the output of `ld -v` on your machine?

Mine is

```
@(#)PROGRAM:ld PROJECT:ld-1053.12
BUILD 15:45:29 Feb  3 2024
configured to support archs: armv6 armv7 armv7s arm64 arm64e arm64_32 i386 x86_64 x86_64h armv6m armv7k armv7m armv7em
will use ld-classic for: armv6 armv7 armv7s arm64_32 i386 armv6m armv7k armv7m armv7em
LTO support using: LLVM version 15.0.0 (static support for 29, runtime is 29)
TAPI support using: Apple TAPI version 15.0.0 (tapi-1500.3.2.2)
```

And compiling the tweak is fine for me.

```
poomsmart@M1-MBP simjectExampleTweak % make
> Making all for tweak simjectExampleTweak…
==> Preprocessing simjectExampleTweak.x…
==> Compiling simjectExampleTweak.x (arm64)…
==> Linking tweak simjectExampleTweak (arm64)…
ld: warning: -multiply_defined is obsolete
ld: warning: -allow_simulator_linking_to_macosx_dylibs is obsolete
==> Generating debug symbols for simjectExampleTweak…
==> Stripping simjectExampleTweak (arm64)…
==> Merging tweak simjectExampleTweak…
==> Signing simjectExampleTweak…
```

Xcode 15.3, macOS Sonoma 14.4

**Comment by newpolygons** [comment]
Getting the same error the output is

> @(#)PROGRAM:ld  PROJECT:dyld-1022.1
BUILD 05:27:37 Dec  7 2023
configured to support archs: armv6 armv7 armv7s arm64 arm64e arm64_32 i386 x86_64 x86_64h
will use ld-classic for: armv6 armv7 armv7s arm64_32 i386 armv6m armv7k armv7m armv7em
LTO support using: LLVM version 15.0.0 (static support for 29, runtime is 29)
TAPI support using: Apple TAPI version 15.0.0 (tapi-1500.0.12.8)
Library search paths:
Framework search paths:


MacOS: 13.6.5
Xcode: 15.2

**Comment by PoomSmart** [comment]
`$THEOS/bin/update-theos`

And your issue will be fixed. :)
